### PR TITLE
test(nakama): Use mockery to create mocks of Nakama interfaces, and write some samp…

### PR DESCRIPTION
**Pull request by jerargus**
…le unit tests demonstrating the functionality.

Closes: WORLD-659

## Overview

Use mockery to generate mocks for Nakama interfaces, and implement some unit tests to demonstrate the functionality.

This PR only has test changes; no business logic source code has been changed.

## Testing and Verifying

Use `go test` to test Nakama.

### CODE REVIEW

File: relay/nakama/allowlist_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"testing"
+
+	"github.com/heroiclabs/nakama-common/api"
+	"github.com/stretchr/testify/mock"
+	"gotest.tools/v3/assert"
+	"pkg.world.dev/world-engine/relay/nakama/mocks"
+)
+
+func TestAllowList_UserIDRequired(t *testing.T) {
+	enableAllowListForTest(t)
```
Comment (technicallyty) [inline relay/nakama/allowlist_test.go:15]: nit: we could make use of a `testify.Suite` and implement the `SetupTest` which would run some logic before each test in the suite.

see example here: https://github.com/Argus-Labs/world-engine/blob/cf1f24695b7d9bfd7d85c225d933843d93470972/evm/precompile/router/router_test.go

File: relay/nakama/allowlist_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	// This context does not have a user ID.
+	ctx := context.Background()
+
+	_, err := claimKeyRPC(ctx, noopLogger(t), nil, nil, "")
+	assert.Check(t, err != nil)
```
[the remark this answers is not in the document]
Comment (jerargus) [reply relay/nakama/allowlist_test.go:20]: This ensures claimKeyRPC fails when no userID is in the context, so this Check is correct.

[the remark this answers is not in the document]
Comment (technicallyty) [reply relay/nakama/allowlist_test.go:20]: oh whoops, had it backwards in my head 🤪

**Comment by jerargus** [comment]
I'm closing this PR. I thought it was going well, but I'm hitting build issues with these new dependencies. I have some stuff to figure out locally.

**Pull request closed without merging**